pydm/display.py

```diff
@@ -104,13 +155,11 @@ def load_ui_file(uifile, macros=None, args=None):
     """
 
     d = Display(macros=macros)
-    if macros:
-        f = macro.substitute_in_file(uifile, macros)
-    else:
-        f = uifile
-
     d._loaded_file = uifile
-    _load_ui_into_display(f, d)
+    code_string, class_name = _compile_ui_file(uifile)
+    if macros:
+        code_string = macro.replace_macros_in_template(Template(code_string), macros).getvalue()
+    _load_compiled_ui_into_display(code_string, class_name, d)
```

Comment:
I'm surprised that this code snippit is duplicated in `Display.load_ui`, it should be possible (but not necessary) to refactor a bit

Reply:
Yes, I will clean that up